### ISSUE TITLE
Performance/Size only works on Arrays/Hashes themselves, not variables

### ISSUE THREAD

**Issue by samlehman** [bbatsov/rubocop#3062]
I grabbed the example code from the Performance/Size cop and added some lines of my own at the bottom:

Test code:
```ruby
# bad
[1, 2, 3].count

# bad
{ a: 1, b: 2, c: 3 }.count

# good
[1, 2, 3].size

# good
{ a: 1, b: 2, c: 3 }.size

# good
[1, 2, 3].count { |e| e > 2 }

# When you try using variables:
arr = [1,2,3]

# should be bad?
arr.count

# should be good?
arr.size
```

## Expected behavior
* It should warn you about using .count on a variable which represents an array or hash (`arr.count`)

## Actual behavior
* Doesn't warn you on `arr.count`

## Steps to reproduce the problem
* save code above to a file
* `rubocop --only Performance/Size blah.rb`

## RuboCop version
* 0.37.2 (using Parser 2.3.0.6, running on ruby 2.1.3 x86_64-darwin13.0)

**Comment by tejasbubane** [comment]
I think this is out of scope of `static analysis`. There is no way to know if the given variable holds an `array` or `hash` or something else unless the code is actually executed.

**Comment by alexdowad** [comment]
In some cases, it is not terribly difficult to determine that `arr` will hold an `Array`, but how far do we go down that road? In most real-world situations, it would be impossible, or *very* difficult.

If you are writing a thesis for your degree and want to do this as a research project, that is fine. I don't think any of the existing contributors have anywhere close the amount of time required to do this well, though.

**Comment by samlehman** [comment]
Ahh ok, thanks @tejasbubane & @alexdowad.
